sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PortEvaluationStage.java

```diff
@@ -158,37 +122,32 @@ protected void setProtos(PodInfoBuilder podInfoBuilder, Protos.Resource resource
         if (getTaskName().isPresent()) {
             String taskName = getTaskName().get();
             Protos.TaskInfo.Builder taskBuilder = podInfoBuilder.getTaskBuilder(taskName);
-            taskBuilder.getCommandBuilder().setEnvironment(
-                    withPortEnvironmentVariable(taskBuilder.getCommandBuilder().getEnvironment(), port));
-
-            if (taskBuilder.hasDiscovery()) {
-                Protos.Ports.Builder portsBuilder = taskBuilder.getDiscoveryBuilder().getPortsBuilder();
-                Optional<Protos.Port.Builder> builderOptional = portsBuilder.getPortsBuilderList().stream()
-                        .filter(b -> b.getName().equals(getPortName()))
-                        .findFirst();
-                if (builderOptional.isPresent()) {
-                    setPortBuilder(builderOptional.get(), port);
-                } else {
-                    portsBuilder.addPorts(makePortBuilder(port));
 
-                }
-            } else {
-                Protos.DiscoveryInfo.Builder discoveryInfoBuilder = Protos.DiscoveryInfo.newBuilder()
-                        .setVisibility(Protos.DiscoveryInfo.Visibility.FRAMEWORK)
+            if (!taskBuilder.hasDiscovery()) {
+                // Initialize with defaults:
+                taskBuilder.getDiscoveryBuilder()
+                        .setVisibility(Constants.DEFAULT_TASK_DISCOVERY_VISIBILITY)
```

Comment:
The default for Marathon is `FRAMEWORK` visibility for ports to generate SRV records. We only increase visibility when there is a VIP on a port. IIUC to replicate Marathon's behavior the logic is:
port has VIP -> EXTERNAL
port does not have vip -> FRAMEWORK

there is a CI test for generating SRV records, so if that passes then this is all moot I suppose, but with noting in case we hit an error later. 

Looks like you have the visibility in the YAML?